homeassistant/components/climate/__init__.py

```diff
@@ -219,6 +237,32 @@ def away_mode_set_service(service):
         descriptions.get(SERVICE_SET_AWAY_MODE),
         schema=SET_AWAY_MODE_SCHEMA)
 
+    def home_mode_set_service(service):
+        """Set home mode on target climate devices."""
+        target_climate = component.extract_from_service(service)
+
+        home_mode = service.data.get(ATTR_HOME_MODE)
+
+        if home_mode is None:
```

Comment:
This will never happen since the schema will validate the passed attributes before this function is called. You can remove this whole block.